core/web/loader/getters.go

```diff
@@ -158,6 +158,24 @@ func GetJobProposalsByFeedsManagerID(ctx context.Context, id string) ([]feeds.Jo
 	return jbRuns, nil
 }
 
+// GetJobByExternalJobID fetches the job proposals by external job ID
+func GetJobByExternalJobID(ctx context.Context, id string) (*job.Job, error) {
+	ldr := For(ctx)
+
+	thunk := ldr.JobsByExternalJobIDs.Load(ctx, dataloader.StringKey(id))
+	result, err := thunk()
+	if err != nil {
+		return nil, err
+	}
+
+	job, ok := result.(job.Job)
+	if !ok {
+		return nil, errors.New("invalid type")
```

Comment:
should we make this error a `var`?  seems like it could be used commonly

Reply:
updated! like this slight clean up suggestion